rfd/0194-selinux-policy.md

```diff
@@ -0,0 +1,150 @@
+be more permissive, defeating the point of the policy in the first place.
+
+If SELinux is not present on the host, the subcommand will tell the user this and exit. If SELinux is present but the Teleport agent policy is
+not installed, the subcommand will extract the embedded policy and use `checkmodule`, `semodule_package`, and `semodule` to compile and install the policy.
+
+Installation scripts and the `teleport-update` tool will be updated to install and configure the SELinux policy with the `selinux` subcommand.
+
+SELinux has 3 modes, disabled, permissive (policy violations are only logged) and enforcing (attempts to violate the policy are denied).
+The SELinux mode will not be modified by the `selinux` subcommand, users will be responsible for enabling SELinux if they so choose.
+
+The `selinux` subcommand will have a `--dry-run` flag that will print what actions it would need to take to manage the policy if run normally and
+print the policy that would be installed based on the provided Teleport agent configuration file.
+
+### Compliance
+
+To ensure the SSH agent SELinux policy is installed and SELinux is configured to enforce it, a `--selinux` flag will be added to the `teleport start` subcommand
+of Teleport agents. If the SELinux policy is not installed or SELinux is not present and in enforcing mode and `--selinux` is passed, the Teleport agent
+will exit with an error.
+
+Additionally since the SSH service is the only Teleport agent service that will be initially supported by the policy, if the `--selinux` flag is passed
+and the Teleport agent does not have the SSH service enabled, or at least one other Teleport service is enabled then the Teleport agent will exit with an error.
+Additionally if there is a SELinux policy for Teleport agents installed but it does not match the policy that would be generated with the version of the
+policy embedded in the Teleport agent and the Teleport agent's configuration file it will notify the user and exit with an error. Users will then have to
+run the `selinux` subcommand to update the SELinux policy.
+
+### Host support
+
+Initially RHEL based distributions, primarily RHEL 7 and 8, and Rocky Linux 8 will be supported.
+Other distributions such as Debian based distributions may be supported in the future.
+
+### Security
+
+This policy when installed and SELinux is set to the enforcing mode will only increase the security of Teleport SSH agents. The authenticity of
+the policies contained in release assets served by the download server is out of scope for this RFD.
+
+### Test plan
+
+Install the policy in permissive mode before running through test cases that involve the SSH service. Check that exercising the test cases do not
+create any policy violations.
+
+Specifically, the following SSH service features will be tested with the policy installed:
```

Comment:
This should also test that some actions are blocked, such as running other commands.